src/debug_toolkit/main.py

```diff
@@ -207,10 +207,14 @@ def memory(pid: int, seconds: int = 60, verbose: bool = False):
     inject_string(pid, payload, trampoline=True, trampoline_timeout=timeout_seconds, verbose=verbose)
 
 @app.command()
-def stack_trace(pid: int, all_threads: bool = True, verbose: bool = False):
+def stack_trace(pid: int, all_threads: bool = True, amount: int = 1, sleep_duration_s: int=0, verbose: bool = False):
+    if amount < 1 or sleep_duration_s < 0:
+        typer.secho(f"ERROR! amount must be greater than 1 and sleep must be greater than 0", fg="red")
     payload = pkgutil.get_data(__package__, "payloads/stack_trace.py").decode()
     payload = payload.replace("ALL_THREADS_PLACEHOLDER", str(all_threads))
-    inject_string(pid, payload, trampoline=True, trampoline_timeout=10, verbose=verbose)
+    payload = payload.replace("AMOUNT_PLACEHOLDER", str(amount))
+    payload = payload.replace("SLEEP_DURATION_S_PLACEHOLDER", str(sleep_duration_s))
+    inject_string(pid, payload, trampoline=True, trampoline_timeout=amount*(sleep_duration_s+10), verbose=verbose)
```

Comment:
Can you explain the `trampoline_timeout` duration calculation? 
If, for example, sleep_duration is 10, and amount is 10. 
trampoline_timeout would be 200?
What exactly is trampoline_timeout used for?

Reply:
The idea behind the calculation is amount*sleep duration is the minimum amount of time each command could take, sleep duration is 10, and the amount is 10 than there will be atleast 100 seconds of sleep. the extra 10 seconds is because it is possible that in some stage it will be stuck waiting for some resource, (even when we do the trampoline or stack trace it can take a few seconds) worse case scenario- so for each trace we have `(sleep time + 10 second worse case command time)`. And then we multiply them by the amount of traces running.
`trampoline_timeout` is used in `do_trampoline_injection`
for each command we run in the toolkit, we poll every second to see if there is an output from the command, if the time is longer than the timeout we return an error.

Reply:
got it. thanks